File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program106.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		.Write a program which accept string from user and accept one character. 	
//		Check whether that character is present in string or not.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt;
	boolean bAns;
	char ch;

	Solution(String str,char ch)
	{
		this.str=str;
		this.iCnt=0;
		this.bAns = false;
		this.ch = ch;
	}

	boolean ChkChar()
	{
		
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)==ch)
			{
				bAns = true;
				break;
			}
			iCnt++;
		}
		
		return bAns;
	}
}

class Program
{
	public static void main(String args[])
	{
		boolean bRet=false;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		System.out.println("Enter the character: ");
		char ch = sObj.next().charAt(0);

		Solution obj = new Solution(str,ch);
		bRet = obj.ChkChar();

		if(bRet==true)
		{
			System.out.println("Character is present");
		}
		else
		{
			System.out.println("Character is absent");
		}
	}
}

/*
OUTPUT:
EEnter the string: 
Vishal Patil
Enter the character: 
a
Character is present

*/

File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program106.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		.Write a program which accept string from user and accept one character. 	
//		Check whether that character is present in string or not.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt;
		bool bAns;
		char ch;

	public:

		Solution(string str,char ch,int iCnt=0,bool bAns=false)
		{
			this->str=str;
			this->ch=ch;
		}

		bool ChkChar()
		{
			while(iCnt!=str.length())
			{
				if(str[iCnt]==ch)
				{
					bAns = true;
					break;
				}

				iCnt++;
			
			}	
			return bAns;
		}
};

int main()
{
	string str;
	char ch='\0';
	bool bRet=false;
	
	cout<<"Enter string: ";
	getline(cin,str);

	cout<<"Enter character: ";
	cin>>ch;
	
	Solution *sObj = new Solution(str,ch);
	bRet = sObj->ChkChar();
	if(bRet==true)
	{
		cout<<"Character is present\n";
	}
	else
	{
		cout<<"Character is absent\n";
	}

	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: Marvellous Mmulti OS
Enter character: e
Character is present

Enter string: Marvellous Multi OS
Enter character: w
Character is absent

*/
